src/libraries/System.Net.WebSockets.Client/tests/DeflateTests.cs

```diff
@@ -86,6 +86,81 @@ public DeflateTests(ITestOutputHelper output) : base(output)
+                using var cts = new CancellationTokenSource(TimeOutMilliseconds);
+
+                cws.Options.DangerousDeflateOptions = deflateOpt;
+                await ConnectAsync(cws, uri, cts.Token);
+
+
+                await cws.SendAsync(Memory<byte>.Empty, WebSocketMessageType.Text, WebSocketMessageFlags.DisableCompression, default);
+                Assert.Throws<ArgumentException>("messageFlags", () =>
+                   cws.SendAsync(Memory<byte>.Empty, WebSocketMessageType.Binary, WebSocketMessageFlags.EndOfMessage, default));
+            }, server => server.AcceptConnectionAsync(async connection =>
+            {
+                var extensionsReply = CreateDeflateOptionsHeader(deflateOpt);
+                await LoopbackHelper.WebSocketHandshakeAsync(connection, extensionsReply);
+            }), new LoopbackServer.Options { WebSocketEndpoint = true });
+        }
+
+        [ConditionalFact(nameof(WebSocketsSupported))]
+        public async Task SendHelloWithDisableCompression()
+        {
+            byte[] bytes = "Hello"u8.ToArray();
+            byte[] compressed = new byte[] { 0xc1, 0x07, 0xf2, 0x48, 0xcd, 0xc9, 0xc9, 0x07, 0x00 };
+            var deflateOpt = new WebSocketDeflateOptions
+            {
+                ClientMaxWindowBits = 14,
+                ClientContextTakeover = true,
+                ServerMaxWindowBits = 14,
+                ServerContextTakeover = true
+            };
+
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
+            {
+                using var cws = new ClientWebSocket();
+                using var cts = new CancellationTokenSource(TimeOutMilliseconds);
+
+                cws.Options.DangerousDeflateOptions = deflateOpt;
+                await ConnectAsync(cws, uri, cts.Token);
+
+                await cws.SendAsync(bytes, WebSocketMessageType.Text, true, cts.Token);
+
+                WebSocketMessageFlags flags = WebSocketMessageFlags.DisableCompression | WebSocketMessageFlags.EndOfMessage;
+                await cws.SendAsync(bytes, WebSocketMessageType.Text, flags, cts.Token);
+            }, server => server.AcceptConnectionAsync(async connection =>
+            {
+                var buffer = new byte[bytes.Length];
+                var extensionsReply = CreateDeflateOptionsHeader(deflateOpt);
+                await LoopbackHelper.WebSocketHandshakeAsync(connection, extensionsReply);
+                using WebSocket websocket = WebSocket.CreateFromStream(connection.Stream, new WebSocketCreationOptions
+                {
+                    IsServer = true,
+                    DangerousDeflateOptions = deflateOpt
+                });
+
+                Assert.True(websocket.State == WebSocketState.Open || websocket.State == WebSocketState.CloseSent);
+
+                await websocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
+                Assert.Equal(bytes, buffer);
```

Comment:
hmmm... we can't see in public api whether the message was compressed or not, can we? (as WebSocket applies decompression on ReceiveAsync) 🤔
can we maybe peek the raw bytes on the stream to be sure?

Reply:
Yes, it seems so. But to be honest I think checking compression is unrelated to the change. Either way I have no chance now to make update before going OOF so feel free to add the check or merge as it is.

Reply:
Discussed offline that I'll take over the PR to update the test